log4j2-propagation/src/main/java/nl/talsmasoftware/context/log4j2/ThreadContextData.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016-2021 Talsma ICT
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.talsmasoftware.context.log4j2;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.ThreadContext;
+
+/**
+ * Represents a snapshot of the data of the Log4j 2 {@link ThreadContext} of a
+ * specific thread at a certain point in the past.
+ */
+public class ThreadContextData {
```

Comment:
When multiple context managers are applied it may be useful to know this is Log4j2's data?
Perhaps create an interface `Log4j2ThreadContext` with implementation `Log4j2ThreadcontextData` ?

Reply:
I am not sure if having an interface is worth it, see https://github.com/talsma-ict/context-propagation/pull/207#discussion_r577189357. We are not expecting the user to provide their own implementation of that class, are we?